fmriprep/workflows/confounds.py

```diff
@@ -0,0 +1,109 @@
+'''
+Workflow for discovering confounds.
+Calculates frame displacement, segment regressors, global regressor, dvars, aCompCor, tCompCor
+'''
+from nipype.interfaces import utility, nilearn
+from nipype.algorithms import confounds
+from nipype.pipeline import engine as pe
+
+from fmriprep.interfaces import mask
+from fmriprep import interfaces
+
+# this should be moved to nipype. Won't do it now bc of slow PR approval there
+# I'm not 100% sure the order is correct
+FAST_DEFAULT_SEGS = ['CSF', 'gray matter', 'white matter']
+
+def discover_wf(settings, name="ConfoundDiscoverer"):
+    ''' All input fields are required.
+
+    Calculates global regressor and tCompCor
+        from motion-corrected fMRI ('inputnode.fmri_file').
+    Calculates DVARS from the fMRI and an EPI brain mask ('inputnode.epi_mask')
+    Calculates frame displacement from MCFLIRT movement parameters ('inputnode.movpar_file')
+    Calculates segment regressors and aCompCor
+        from the fMRI and a white matter/gray matter/CSF segmentation ('inputnode.t1_seg')
+
+    Saves the confounds in a file ('outputnode.confounds_file')'''
+
+    inputnode = pe.Node(utility.IdentityInterface(fields=['fmri_file', 'movpar_file', 't1_seg',
+                                                          'epi_mask']),
+                        name='inputnode')
+    outputnode = pe.Node(utility.IdentityInterface(fields=['confounds_file']),
+                         name='outputnode')
+
+    # Global and segment regressors
+    signals = pe.Node(nilearn.SignalExtraction(include_global=True, detrend=True,
```

Comment:
oh wow. that is supposed to be `nipype.SignalExtraction`. how did that even work?